### ISSUE TITLE
Replace hide=resources for count purposes

### ISSUE THREAD

**Issue by chrisarcand** [ManageIQ/manageiq-api#209]
We currently use `?hide=resources` to hide href's in collections for the purposes of reducing the response when a client only cares about the count of the resources.

Instead of doing this, a more conventional, semantically proper, and performant way to do this is by either including the total count in the headers of the collection request whereby a client can then just request `HEAD /resources` to return the count without a response body at all, or simply add a proper route with only the count as a response (`/resources/count`)

I propose the former, using HTTP semantics is a good thing.

> The HTTP HEAD method requests the headers that are returned if the specified resource would be requested with an HTTP GET method. Such a request can be done before deciding to download a large resource to save bandwidth, for example.

A header such as `x-resource-count` or whatever should be added to collections and clients such as the SUI should be using the header from a `HEAD` request.

**Comment by chrisarcand** [comment]
cc/ @imtayadeway @AllenBW from https://github.com/ManageIQ/manageiq/pull/12694

**Comment by AllenBW** [comment]
We *DO* use them quite often for sidebar counts.  I'm down for whatever!  If a change needs to be made lemme know and it'll be done

Ok now to the questions: 
1. Are you saying that as a header or a proper route the performance of this call will have a lesser impact on server load?
2. Are you proposing the depreciation of `?hide=resources `? 
3. Custom  headers as a means of grabbing metadata are a little surprising, not something ya might know unless explicitly told, does this work include creating the documentation to say, "hey, yeah, we do this now, grabbing skinny metadata with custom headers"?

**Comment by AllenBW** [comment]
RE: and yah, `?hide=resources` is used when we only want the count of a query, none of the resources, actions, or paging

**Comment by chrisarcand** [comment]
> Are you saying that as a header or a proper route the performance of this call will have a lesser impact on server load?

Yes; we need only respond with a count when all you're asking for is a count and optimize for that case, and while the `hide=resources` is a slight optimization this would be a much better one - and make more sense semantically. It's sort of a weird shoehorning of something that doesn't imply it's purpose in the querystring besides.

> Are you proposing the depreciation of ?hide=resources ?

Yes; this is it's sole purpose (to my knowledge) and this new way would replace it. 

> Custom headers as a means of grabbing metadata are a little surprising, not something ya might know unless explicitly told, does this work include creating the documentation to say, "hey, yeah, we do this now, grabbing skinny metadata with custom headers"?

Surprising only if it's not documented, yeah 😄  Yes, I'll change the site documentation accordingly to help make it obvious that this is the correct way to do it.

> RE: and yah, ?hide=resources is used when we only want the count of a query, none of the resources, actions, or paging

Excellent, that's what I figured.

---

Re: question 3, I'd like to discuss a reimplementation of API documentation to help consumers of the API like you to make sense of what's being deprecated, *when* it's being deprecated ("What release was this available? Deprecated? What changed?), etc etc. We currently don't have any way to deduce that from the documentation at all and I can respect from your end that it's probably pretty annoying to deal with. This is a whole separate can of worms that I'll bring up later, but thought I'd mention here as this is a *great* example.

**Comment by Fryguy** [comment]
>  I'd like to discuss a reimplementation of API documentation

This should probably be discussed in the context of our discussion about moving to swagger, since the way the swagger API is documentation driven.

**Comment by abellotti** [comment]
👍 on swagger, @mkanoor did a nice presentation/poc on that.

**Comment by chrisarcand** [comment]
Yup!

**Comment by JPrause** [comment]
@abellotti is this still a valid issue? If yes, please remove the stale label. If not can you close.
If there's no update by next week, I'll be closing this issue.

**Comment by JPrause** [comment]
Closing issue. If you feel the issue needs to remain open, please either reopen or let me know and it will be reopened. 
@miq-bot close_issue
